ui/redux/actions/publish.js

```diff
@@ -633,7 +639,35 @@ export const doPublish = (success: Function, fail: Function, preview: Function,
   // get redux publish form
   const publishData = selectPublishFormValues(state);
 
+  /** protected content and members-only chat functionality **/
+  const { restrictCommentsAndChat, restrictedToMemberships, channelClaimId } = publishData;
+
   const publishPayload = payload || resolvePublishPayload(publishData, myClaimForUri, myChannels, preview);
+  publishPayload.tags = new Set(publishPayload.tags);
+
+  const publishTagsHasRestrictedMemberships = publishPayload.tags.has(MEMBERS_ONLY_CONTENT_TAG);
+  const publishTagsHasRestrictedChatComments = publishPayload.tags.has(RESTRICTED_CHAT_COMMENTS_TAG);
+
+  // add members only tag if it's restricted to memberships and tag doesn't exist
+  if (restrictedToMemberships && !publishTagsHasRestrictedMemberships) {
+    publishPayload.tags.add(MEMBERS_ONLY_CONTENT_TAG);
+  }
+
+  if (restrictCommentsAndChat && !publishTagsHasRestrictedChatComments) {
+    publishPayload.tags.add(RESTRICTED_CHAT_COMMENTS_TAG);
+  }
+
+  if (!restrictedToMemberships && publishTagsHasRestrictedMemberships) {
+    publishPayload.tags = publishPayload.tags.delete(MEMBERS_ONLY_CONTENT_TAG);
+  }
+
+  if (!restrictCommentsAndChat && publishTagsHasRestrictedChatComments) {
+    publishPayload.tags = publishPayload.tags.delete(MEMBERS_ONLY_CONTENT_TAG);
+  }
+
+  publishPayload.tags = Array.from(publishPayload.tags);
+
+  // return
```

Comment:
:small_blue_diamond: This should be moved into `resolvePublishPayload` -- that should be the only place to determine the payload.
I don't think we want/should recalculate on a resumed-publish case.